libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -104,6 +104,56 @@ string YulUtilFunctions::copyToMemoryFunction(bool _fromCalldata)
 	});
 }
 
+string YulUtilFunctions::requireOrAssertFunction(bool _assert, Type const* _messageType)
+{
+	string functionName =
+		string(_assert ? "do_assert" : "require") +
+		(_messageType ? "_withMessage" : "");
```

Comment:
Hm, I'm not 100% sure, but it would probably be better to also encode `_messageType->identifier()` into the function name.